personalities/sctfe/config.go

```diff
@@ -122,7 +122,7 @@ func LogConfigFromFile(filename string) (*configpb.LogConfig, error) {
 //   - Merge delays (if present) are correct.
 //
 // Returns the validated structures (useful to avoid double validation).
-func ValidateLogConfig(cfg *configpb.LogConfig, origin string, projectID string, bucket string, spannerDB string, rootsPemFile string) (*ValidatedLogConfig, error) {
+func ValidateLogConfig(cfg *configpb.LogConfig, origin string, projectID string, bucket string, spannerDB string, rootsPemFile string, rejectExpired bool, rejectUnexpired bool, extKeyUsages string, rejectExtensions string) (*ValidatedLogConfig, error) {
```

Comment:
Is the plan eventually to have a non-proto config struct which gets passed in here with all the required info inside?

Reply:
Eventually yes. I would like:
 - `main.go` to communicate with sctfe libraries with a struct, rather than passing all these arguments.
 - `main.go` to be the one instantiating storages
 - to simplify the layers of structs in the sctfe libarary

There's a bunch of layers, and I'm trying to remove layers one after the other. The first thing I'm trying to do, is to get rid of `cfg *configpb.LogConfig` totally, and pass everything as an argument to `ValidateConfig`.  This is not  very pretty, not the end state, and will certainly change after this, I'm still trying a few things out to find a good shape. 

For now, as an intermediary step, I'm thinking of `ValidateConfig` being more of a `ParseFlag` function / sanitize flags function, to make sure that fields are not empty and that dates don't step on each other for instance. Right now, it also parses the ExtendedKeyUsage, I might move it as well. I don't really know how much logic I want in there.

Then there's `SetUpInstance` which actually applies some logic on stuff and open the root_pem_file, or parses the oids. Logically, I think that parsing the oids is similar to parsing the ExtendedKeyUsage, so I might group them together.

Also, right now, the Tessera storage cannot be created anywhere else than in `SetupInstance` because, it needs a signer, and the signer is created in there... because we need a note.Signer for the checkpoint, but not for the SCTs.

Then theres's `newLogInfo`, and this one is a bit annoying as well because it's the one that as the TimeSource object, which is also used in tests.